test/case/ietf_system/ssh_key_authentication/test.py

```diff
@@ -0,0 +1,100 @@
+Bran5mNdcGzf/8SD25WRMUPhsoqHOvtDWuOkop6nAu/9GGigJaBICuDxJz3KN1Hhpuer+g
+/x7+Dgwd5mW41hXOTezp2hGO98OwHLJpmohCz1NIZIHz0h6NZbRZ4vozMtYBs1aq1Qv6h2
+DLsxLMLY7FAAAAgQDKkDQXxiXr7Bv7UJaYA4BZAg6qPMvW943mOreLl0uO8urE3mt2MnVy
+Et1r2dEhxwqsWrDj5rsBVSXrg/8kXYHrqAgldZ3TxEOCU8xVV4RAzxfE0uFwdawZHdsIJL
+JWwNnQ8TdWXPZze0lE9R5b2E4EvJilHvD13pdcIN+hyvGE0QAAAIEAyY+JAtJjDS6wW07E
+eTVndLHpDiTZO+tCIP+OO45uYA/8O+IMc+wvIsHCpZC7e3bskg6z2gt5B0DwSnf2Q4zXxP
+OFXdOBTFO8XcgWKRo9BIPV6BNH9qrx0Z0m5G45rY6SE9c5Ypv0ExjXRZ/iPWLBSarsv1fF
+lrH5aNT6hzZKsc8AAAAMcm9vdEBpbmZhbXkwAQIDBAUGBw==
+-----END OPENSSH PRIVATE KEY-----
+"""       
+        
+        public_key_alg = "ssh-rsa"
+        public_key_data = "AAAAB3NzaC1yc2EAAAADAQABAAABAQCffMvdK1VBkuw/12etfEr89Piba2NbQz7YY3ol2IhRnIoj0oodZp/a4c11c1Q+jQkbCDKPWB45w3ZMR2Q93XXeY72qtLzvCBLOzeCkX3czMAJLLxznl3EXmDnTAWetE1nUmByLLOvKdw1CPFiS9tushwdfj6iOfex3HvQwltV5i/rQz0h7JeSwEQ0Xn6sBgDxWlD6TtwYl4ch0zHoxRHCjqdQig7NMonhaF53BMKRYaUSLET3w+zKfK8HYiHxry6Jf+dcts9PQyxpCiEW3RoU65nwnq1pG7XAdmT/DeGKSQqXkdnmXvgZkrQYZ+4HW2veQlHzwwEIAhMTscS+QiOX/"
+
+    with test.step("Create a guest user on the target device"):
+        USER = "guest"
+
+        target.put_config_dict("ietf-system", {
+            "system": {
+                "authentication": {
+                    "user": [
+                        {
+                            "name": USER,
+                            "authorized-key": [
+                                {
+                                    "name":"guest-ssh-key",
+                                    "algorithm":public_key_alg,
+                                    "key-data":public_key_data
+                                }
+                            ],
+                            "infix-system:shell":"bash"
+                        }
+                    ]
+                }
+            }
+        })
+    
+    with test.step("Write private key to a temporary file"):
+        with tempfile.NamedTemporaryFile(delete=False, mode='w') as key_file:
+            key_file.write(private_key)
+            key_file_name = key_file.name
+
+        os.chmod(key_file_name, 0o600)
+
+    with test.step("Verify it is possible to fetch syslog data using public key"):
+        address = target.get_mgmt_ip()
+        syslog_file = "./syslog_copy"
+        result = subprocess.run(f"scp -q -o StrictHostKeyChecking=no -i {key_file_name} {USER}@[{address}]:/var/log/syslog {syslog_file}", shell=True)
+
+        if result.returncode != 0:
+            test.fail("Failed to copy syslog from target device")
+
+    with test.step("Clean up temporary files"):
+        try:
+            os.remove(key_file_name)
+        except OSError as e:
+            print(f"Error removing temporary key file: {e}")
+
+        if os.path.exists(syslog_file):
+            try:
+                os.remove(syslog_file)
+            except OSError as e:
+                print(f"Error removing syslog file: {e}")
```

Comment:
With a refactor, this whole section could be moved/hidden away from the user. E.g., an API *could* look like this:

```python
    def ssh.fetch_file(name, check=False, remove=False):
       ..
```

Where check does the above check (previous comment), and remove encapsulates this current code path.  (For a generic function you might not want the API to remove the file you fetched ;-)